docs/modules/ROOT/pages/explanations/decisions/user-management.adoc

```diff
@@ -0,0 +1,200 @@
+
+Advantages::
+* Each service has exactly one source of truth
+* Simple to understand for the end-user, permissions and security can be defined right in the claim
+
+Disadvantages::
+* A claim could get very large
+
+=== Proposal 2: User management separate of claims
+
+Each service has additional XRDs that manage the user and permissions for the given service.
+
+There are two variations how this could be done.
+Having a reference to the instance in the user management XRD.
+Or Having a reference for the user management XRD in the main claim.
+
+The complexity of this solution will be much higher:
+
+* Each service needs a new XRD
+* Each of these XRDs need a new composition
+* Each of these XRDs need a new composition function
+* The function itself first needs to resolve the references before it can actually deploy the necessary configurations for the provider
+
+Example reference on management object:
+
+[source,yaml]
+----
+apiVersion: vshn.appcat.vshn.io/v1
+kind: VSHNPostgreSQL
+metadata:
+  name: my-pg
+spec:
+  parameters: {}
+---
+apiVersion: vshn.appcat.vshn.io/v1
+kind: VSHNPostgreSQLIAM
+metadata:
+  name: myiam
+spec:
+  parameters:
+    postgreSQLClaimRef:
+      name: my-pg
+    service:
+      users: <1>
+        - prod
+        - test
+        - int
+      databases: <2>
+        - prod
+        - int
+        - test
+      grants: <3>
+        - privileges:
+            - ALL
+          type: GRANT
+          user: prod
+          database: prod
+----
+
+<1> Generic for PostgreSQL, MariaDB and potentially more
+<2> Generic for PostgreSQL and MariaDB
+<3> PostgreSQL specific, modelled after `provider-sql` 's https://github.com/crossplane-contrib/provider-sql/blob/master/examples/postgresql/grant.yaml#L5[CRDs].
+
+Example reference on claim:
+
+[source,yaml]
+----
+apiVersion: vshn.appcat.vshn.io/v1
+kind: VSHNPostgreSQL
+metadata:
+  name: my-pg
+spec:
+  parameters:
+    service:
+      iamRefs:
+        - myiam
+---
+apiVersion: vshn.appcat.vshn.io/v1
+kind: VSHNPostgreSQLIAM
+metadata:
+  name: myiam
+spec:
+  parameters:
+    users: <1>
+      - prod
+      - test
+      - int
+    databases: <2>
+      - prod
+      - int
+      - test
+    grants: <3>
+      - privileges:
+          - ALL
+        type: GRANT
+        user: prod
+        database: prod
+----
+
+<1> Generic for PostgreSQL, MariaDB and potentially more
+<2> Generic for PostgreSQL and MariaDB
+<3> PostgreSQL specific, modelled after `provider-sql` 's https://github.com/crossplane-contrib/provider-sql/blob/master/examples/postgresql/grant.yaml#L5[CRDs].
+
+Advantages::
+* The claims don't get inflated
+
+Disadvantages::
+* More room for the customer to make mistakes by referencing wrong instances
+* Multiple claims reconcile the same service, no single point of truth
+* Complexity increases a lot
+* Deletion can get messy, what happens if the claim gets deleted, but the users, databases and grant objects still exist?
+
+== Decision
+
+Proposal 1.
```

Comment:
This would mean this proposal isn't multi-tenant self-service aware. It wouldn't be possible to have a cluster-wide PostgreSQL and, for example, a service in a namespace could request access to such a shared database instance.

It would couple instance management and access management, which I'm not sure if this is a suitable approach in general, putting too much domain into a single entity.

Reply:
As discussed in the chat, I will adjust the description that this decision is not about multitenancy systems.